cloud/gcp/deploy/iam/serviceaccount.go

```diff
@@ -0,0 +1,71 @@
+// Copyright Nitric Pty Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iam
+
+import (
+	"github.com/nitrictech/nitric/cloud/azure/deploy/utils"
+	"github.com/pulumi/pulumi-gcp/sdk/v6/go/gcp/serviceaccount"
+	"github.com/pulumi/pulumi-random/sdk/v4/go/random"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+type GcpIamServiceAccount struct {
+	pulumi.ResourceState
+
+	Name string
+
+	ServiceAccount *serviceaccount.Account
+}
+
+type GcpIamServiceAccountArgs struct {
+	AccountId string
+}
+
+// Create a new GCP IAM service account
+func NewServiceAccount(ctx *pulumi.Context, name string, args *GcpIamServiceAccountArgs, opts ...pulumi.ResourceOption) (*GcpIamServiceAccount, error) {
+	res := &GcpIamServiceAccount{
+		Name: name,
+	}
+
+	err := ctx.RegisterComponentResource("nitric:exec:GCPCloudRunner", name, res, opts...)
```

Comment:
Should this be `nitric:exec:GCPCloudRunner` or its own custom resource? If it should be attached to the cloud runner can it just be passed in as a parent in the `opts`?

Reply:
fixed